specifyweb/workbench/upload/upload_table.py

```diff
@@ -55,6 +56,28 @@ def to_json(self) -> Dict:
 
     def unparse(self) -> Dict:
         return { 'baseTableName': self.name, 'uploadable': self.to_json() }
+    
+class DeferredScopeUploadTable(NamedTuple):
+    name: str
+    overrideScope: Optional[Dict[str, Optional[int]]]
+    wbcols: Dict[str, ColumnOptions]
+    static: Dict[str, Any]
+    toOne: Dict[str, Uploadable]
+    toMany: Dict[str, List[ToManyRecord]]
+
+    related_key: str
+    relationship_name: str
+    filter_field: str
+
+    def apply_scoping(self, collection, deffer: bool = True) -> "ScopedUploadTable":
+        if not deffer:
+            from .scoping import apply_scoping_to_uploadtable as apply_scoping
```

Comment:
can you tell me what's the difference between doing imports inline vs at the top of the file?
is there a performance impact?
is it just for convenience?
is it a best practice?

Reply:
These imports are inline to prevent a circular import dependency. 

Because `scoping.py` has top_level/global imports from `upload_table.py`, if these imports were included at the top of `upload_table.py`, whenever scoping.py or upload.py would be 'loaded', it would go through its import statements and try and load the other python file--- which would require it to load the original file, etc...